src/NetscriptJSEvaluator.ts

```diff
@@ -83,8 +83,26 @@ function generateLoadedModule(script: Script, scripts: Map<ScriptFilePath, Scrip
     return script.mod;
   }
 
+  let scriptCode;
+  const fileType = getFileType(script.filename);
+  switch (fileType) {
+    case FileType.JS:
+      scriptCode = script.code;
+      break;
+    case FileType.JSX:
+    case FileType.TS:
+    case FileType.TSX:
+      scriptCode = transformScript(script.filename, script.code, fileType);
+      break;
+    default:
+      throw new Error(`Invalid file type: ${fileType}. Filename: ${script.filename}, server: ${script.server}.`);
+  }
+  if (!scriptCode) {
+    throw new Error(`Cannot transform script. Filename: ${script.filename}, server: ${script.server}.`);
+  }
+
   // Inspired by: https://stackoverflow.com/a/43834063/91401
-  const ast = parse(script.code, { sourceType: "module", ecmaVersion: "latest", ranges: true });
+  const ast = parse(scriptCode, { sourceType: "module", ecmaVersion: "latest", ranges: true });
```

Comment:
So here, you are transforming before parsing. Why not do the same in the RAM calculation? I feel like it would save a lot of dependency/infrastructure headaches.